website/source/docs/auth/oci.html.md

```diff
@@ -0,0 +1,209 @@
+
+    * Run the command to create the policy.
+
+    ```        
+        curl \       
+        --header "X-Vault-Token: $roottoken" \       
+        --request PUT \       
+        --data @vaultadminpolicy.json \       
+        http://[127.0.0.1:8200/v1/sys/policy/vaultadminpolicy](127.0.0.1:8200/v1/sys/policy/vaultadminpolicy)
+    ```
+
+1.  Configure your home tenancy in the vault, so that only users or instances from your tenancy will be allowed to log into vault through the OCI Auth plugin.  
+    * Create a file named hometenancyid.json with the below content, using the tenancy OCID. To find your tenancy OCID, see [https://docs.cloud.oracle.com/iaas/Content/General/Concepts/identifiers.htm](https://docs.cloud.oracle.com/iaas/Content/General/Concepts/identifiers.htm).       
+
+        `{"configValue":"your tenancy ocid here"}`
+        
+    * Configure the homeTenancy parameter in the vault.
+       
+    ```     curl --header "X-Vault-Token: $roottoken" --request PUT \       
+        --data @hometenancyid.json \       
+        http://[127.0.0.1:8200/v1/auth/oci/config/homeTenancyId](127.0.0.1:8200/v1/auth/oci/config/homeTenancyId)
+    ```
+       
+    * Create a vault administrator role in the OCI Auth plugin. This role allows the administrator of the vault to log into vault and grants them the permissions allowed in the policy.
+    * Create a file named vaultadminrole.json with the below contents. The TTL specifies how long, in seconds, tokens for this role are valid.        
+        `{"add_policy_list":"vaultadminpolicy","description":"VaultAdminRole","ttl":"1500"}`
+        
+    * Run the command to create the vault admin role.
+            
+    ```     
+        curl --header "X-Vault-Token: $roottoken" --request PUT \
+        --data @vaultadminrole.json \
+        http://[127.0.0.1:8200/v1/auth/oci/role/vaultadminrole](127.0.0.1:8200/v1/auth/oci/role/vaultadminrole)
+    ``` 
+          
+1.  Add the group or dynamic group OCID corresponding to users or instances to the vault admin role.
+    * Create a file named vaultadminids.json with the below content. The OCID list is a comma-separated list of group and dynamic group OCIDs in your tenancy.       
+        * For a test in dev mode, you can add the OCID of the dynamic group previously created.
+        * In production, add only the OCID of groups and dynamic groups that can take the admin role in vault.
+        
+        `{"add_ocid_list":"[ocid1.group.oc1..dummy1,ocid1.dynamicgroup.oc1..dummy1](ocid1.group.oc1..dummy1,ocid1.dynamicgroup.oc1..dummy1)"}`
+        
+    *  Run the command to add the admin IDs to the vault admin role.        
+
+    ```     
+        curl --header "X-Vault-Token: $roottoken" --request PUT \       
+        --data @vaultadminids.json \  
+        http://[127.0.0.1:8200/v1/auth/oci/role/vaultadminrole](127.0.0.1:8200/v1/auth/oci/role/vaultadminrole)
+    ```            
+
+1.  Log into the vault using instance principal.
```

Comment:
@vuhpham-oracle Can you provide an example of how to login _without_ the CLI helper? It's not clear how the instance metadata is retrieved and then packaged into a Vault login request.

Reply:
Hi @kalafut, @mauruthig will be taking care of updating this documentation.

@mauruthig Could you please answer @kalafut?

Reply:
Done